protoduneana/verticaldrift/singlehit/MCsingleHitCheater_module.cc

```diff
@@ -0,0 +1,287 @@
+#include "art/Framework/Principal/SubRun.h"
+#include "canvas/Utilities/InputTag.h"
+#include "fhiclcpp/ParameterSet.h"
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+#include "art_root_io/TFileService.h"
+
+//LArSoft
+#include "larcore/CoreUtils/ServiceUtil.h"
+#include "lardata/DetectorInfoServices/DetectorPropertiesService.h"
+#include "lardata/DetectorInfoServices/ServicePack.h" 
+#include "larcore/Geometry/Geometry.h"
+#include "lardataobj/RecoBase/Hit.h"
+#include "lardataobj/RecoBase/Track.h"
+#include "lardataobj/RecoBase/TrackHitMeta.h"
+#include "lardataobj/RecoBase/SpacePoint.h"
+#include "larsim/MCCheater/BackTrackerService.h"
+#include "larsim/MCCheater/PhotonBackTrackerService.h"
+#include "larsim/MCCheater/ParticleInventoryService.h"
+#include "larevt/SpaceCharge/SpaceCharge.h"
+#include "larevt/SpaceChargeServices/SpaceChargeService.h"
+#include "nusimdata/SimulationBase/MCTruth.h"
+#include "nusimdata/SimulationBase/MCParticle.h"
+
+// ROOT
+#include "Math/ProbFunc.h"
+#include "TStyle.h"
+#include "TTree.h"
+#include "TGraphErrors.h"
+#include "TF1.h"
+#include "TH1.h"
+#include "TH2.h"
+#include "TLatex.h"
+#include "TCanvas.h"
+#include "TPaveStats.h"
+
+namespace pdvdana 
+{
+  using std::vector;
+  using std::string;
+  class MCsingleHitCheater;
+}
+
+
+class pdvdana::MCsingleHitCheater : public art::EDAnalyzer {
+public:
+  explicit MCsingleHitCheater(fhicl::ParameterSet const& p);
+  // The compiler-generated destructor is fine for non-base
+  // classes without bare pointers or other resource use.
+
+  // Plugins should not be copied or assigned.
+  MCsingleHitCheater(MCsingleHitCheater const&) = delete;
+  MCsingleHitCheater(MCsingleHitCheater&&) = delete;
+  MCsingleHitCheater& operator=(MCsingleHitCheater const&) = delete;
+  MCsingleHitCheater& operator=(MCsingleHitCheater&&) = delete;
+
+  // Required functions.
+  void analyze(art::Event const& e) override;
+
+  bool Inside( std::string k , std::vector<std::string> v);
+  std::vector<std::string> HowManyLabels( std::vector<std::pair<int, std::string>> vTrackIdToLabelPair);
+
+  // Selected optional functions.
+  void beginJob() override;
+  void endJob() override;
+
+private:
+
+  // Declare member data here.
+
+  unsigned int fEventID = 0;
+  const geo::Geometry* fGeom;
+
+  //Input variables
+  std::string fHitLabel , fG4Label;
+  int LogLevel;
+  // working variables
+ 
+
+};
+
+
+pdvdana::MCsingleHitCheater::MCsingleHitCheater(fhicl::ParameterSet const& p)
+  : EDAnalyzer{p},
+  fHitLabel(p.get<std::string>("HitLabel")),
+  fG4Label(p.get<std::string>("G4Label")),
+  LogLevel(p.get<int>("LogLevel"))
+
+{
+  // Call appropriate consumes<>() for any products to be retrieved by this module.
+  fGeom    = &*art::ServiceHandle<geo::Geometry>();
+  auto const clockData = art::ServiceHandle<detinfo::DetectorClocksService const>()->DataForJob();
+  auto const detProp = art::ServiceHandle<detinfo::DetectorPropertiesService const>()->DataForJob(clockData);
```

Comment:
This might set off Clang when we try to cut a release.  `detProp` looks like an unused variable to me.  Can you remove it without changing what the code does?